linux/mcompile/keymap/keymap.cpp

```diff
@@ -4,24 +4,23 @@
 #include <xkbcommon/xkbcommon.h>
 
 
-int convert_WinShiftstate_to_LinuxShiftstate(int win_ShiftState) {
-  if (win_ShiftState == 0)
-    return 0;
-  else if (win_ShiftState == K_SHIFTFLAG)
-    return XCB_MOD_MASK_SHIFT;
-  else if (win_ShiftState == (LCTRLFLAG | RALTFLAG))
-    return XCB_MOD_MASK_LOCK;
-  else if (win_ShiftState == (K_SHIFTFLAG | LCTRLFLAG | RALTFLAG))
-    return (XCB_MOD_MASK_SHIFT | XCB_MOD_MASK_LOCK);
-  else
-    return win_ShiftState;
+int convert_Shiftstate_to_LinuxShiftstate(int shiftState) {
+  // if shiftState is a windows ShiftState: convert the windows ShiftState (0,16,9,25) to a Linux ShiftState (0,1,2,3) that is then used as "Level" in gdk
+  // if shiftState is NOT a windows ShiftState (then in_ShiftState is already a Linux shiftstate): return the entered shiftstate
+
+  if (shiftState == 0)                                           return 0;                                           // Win ss 0  -> Lin ss 0
+  else if (shiftState == K_SHIFTFLAG)                            return XCB_MOD_MASK_SHIFT;                          // Win ss 16 -> Lin ss 1
+  else if (shiftState == (LCTRLFLAG | RALTFLAG))                 return XCB_MOD_MASK_LOCK;                           // Win ss 9  -> Lin ss 2
+  else if (shiftState == (K_SHIFTFLAG | LCTRLFLAG | RALTFLAG))   return (XCB_MOD_MASK_SHIFT | XCB_MOD_MASK_LOCK);    // Win ss 25 -> Lin ss 3
+  else                                                           return shiftState;                                  // Lin ss x  -> Lin ss x
 }
 
-bool ensureValidInputForKeyboardTranslation(int shiftstate, int count, int keycode) {
-  if (shiftstate > count)
+
+bool ensureValidInputForKeyboardTranslation(int gdk_level, gint count, gint keycode) {
+  if (gdk_level >=  (int) count)
     return false;
 
-  if (keycode > (int)keycode_max)
+  if ((int) keycode > (int)keycode_max)
```

Comment:
```suggestion
  if (keycode > (gint)keycode_max)
```

or even better: define `keyaocde_max` as of type `gint` instead of `KMX_DWORD`.

Reply:
OK-defined as gint
" if (gdk_level **>** (int) count ) "    changed again;

Reply:
> " if (gdk_level > (int) count ) " changed again;

why? Maybe I miss something, but for me this looks wrong.